core/core-dumper.el

```diff
@@ -1,204 +0,0 @@
-;; GNU General Public License for more details.
-;;
-;; You should have received a copy of the GNU General Public License
-;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
-
-
-(require 'spinner)
-
-(defvar spacemacs-dump-mode 'not-dumped
-  "Spacemacs dump mode, can be `not-dumped', `dumped' or `dumping'.")
-
-(defvar spacemacs-dump-process nil
-  "The process object to dump Emacs.")
-
-(defvar spacemacs-dump-delayed-functions '()
-  "List of function to execute once the dump file has been loaded.")
-
-(defvar spacemacs-dump-load-path nil
-  "Variable to backup the `load-path' variable built during the dump creation.")
-
-(defvar spacemacs-dump-spinner nil
-  "Spinner curently being displayed on the `global-mode-string'.")
-
-(defvar spacemacs--dump-old-global-mode-string nil
-  "Copy of `global-mode-string' before the spinner is created")
-
-(defconst spacemacs-dump-directory
-  (concat spacemacs-cache-directory "dumps/"))
-
-(defconst spacemacs-dump-buffer-name "*spacemacs-dumper*")
-
-(defconst spacemacs--dump-spinner-construct
-  '("" (:eval (spinner-print spacemacs-dump-spinner))))
-
-(defmacro spacemacs|when-dumping-strict (&rest body)
-  "Execute body if we are really dumping.
-You should not used this function, it is reserved for some specific process."
-  (declare (indent defun))
-  `(when (eq 'dumping spacemacs-dump-mode)
-     ,@body))
-
-(defmacro spacemacs|require-when-dumping (&rest args)
-  "Require feature if dumping."
-  (spacemacs|when-dumping-strict `(require ,@args)))
-
-(defmacro spacemacs|when-dumping (&rest body)
-  "Execute body if dumping.
-This function considers that we are always dumping if dumping is not supported.
-You should always use this function."
-  (declare (indent defun))
-  `(when (not (eq 'dumped spacemacs-dump-mode))
-     ,@body))
-
-(defmacro spacemacs|unless-dumping (&rest body)
-  "Execute body if not dumping"
-  (declare (indent defun))
-  `(unless (eq 'dumping spacemacs-dump-mode)
-     ,@body))
-
-(defmacro spacemacs|unless-dumping-and-eval-after-loaded-dump (funcname &rest body)
-  "Execute BODY if not dumping, delay BODY evaluation after the dump is loaded.
-FUNCNAME is the name of the function that will be created and evaluated at
-the end of the loading of the dump file."
-  (declare (indent defun))
-  (if (eq 'dumping spacemacs-dump-mode)
-      (let ((funcname2 (intern (format "spacemacs//after-dump-%S" funcname))))
-        `(progn
-           (defun ,funcname2 nil ,@body)
-           (add-to-list 'spacemacs-dump-delayed-functions ',funcname2)))
-    `(progn ,@body)))
-
-(defun spacemacs/dump-save-load-path ()
-  "Save `load-path' variable."
-  (setq spacemacs-dump-load-path load-path))
-
-(defun spacemacs/dump-restore-load-path ()
-  "Restore the `load-path' variable from the dump. "
-  (spacemacs|unless-dumping
-    (when (not (null spacemacs-dump-load-path))
-      (setq load-path spacemacs-dump-load-path))))
-
-(defun spacemacs/defer (&optional idle-time)
```

Comment:
This function is used in lots of layers.  We need to replace its callsites (probably mostly with `t`?).  In my config at least, there is an error in the `use-package` for `recentf`.

Reply:
You're right, my latest push had replace them to `t`. Thanks!